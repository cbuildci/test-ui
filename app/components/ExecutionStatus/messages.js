/*
 * ExecutionStatus Messages
 */

import { defineMessages } from 'react-intl';

export default defineMessages({
    executionStatus: {
        id: 'app.components.ExecutionStatus.executionStatus',
        defaultMessage: `{status, select,
            QUEUED {Queued}
            IN_PROGRESS {In Progress}
            SUCCEEDED {Succeeded}
            ERROR {Internal System Error}
            FAILED {Failed}
            STOPPED {Stopped by User}
            TIMED_OUT {Timed Out}
            NEUTRAL {Neutral}
            other {{status}}
        }`,
    },
});
