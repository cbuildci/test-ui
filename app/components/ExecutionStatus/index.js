/**
 * ExecutionStatus
 */

import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import messages from './messages';

export const statusDisplay = {
    SUCCEEDED: { icon: 'fas fa-check', color: '#28a745' },
    QUEUED: { icon: 'far fa-clock', color: '#6c757d' },
    IN_PROGRESS: { icon: 'fas fa-hourglass-start', color: '#17a2b8' },
    ERROR: { icon: 'fas fa-exclamation-triangle', color: '#dc3545' },
    FAILED: { icon: 'fas fa-times', color: '#dc3545' },
    TIMED_OUT: { icon: 'fas fa-stopwatch', color: '#dc3545' },
    STOPPED: { icon: 'fas fa-stop-circle', color: '#dc3545' },
    NEUTRAL: { icon: 'fas fa-share', color: '#6c757d' },
    '': { icon: 'fas fa-question-circle', color: '#333333' },
};

export function getStatusColor(status) {
    return (statusDisplay[status] || statusDisplay['']).color || null;
}

function ExecutionStatus({ status }) {
    const { icon, color } = statusDisplay[status] || statusDisplay[''];

    return (
        <span style={{ color }}>
            {icon && <i className={`${icon} fa-fw mr-1`}/>}
            {status ? <FormattedMessage
                {...messages.executionStatus}
                values={{ status: status }}
            /> : '-'}
        </span>
    );
}

ExecutionStatus.propTypes = {
    status: PropTypes.string,
};

ExecutionStatus.defaultProps = {
    status: null,
};

export default ExecutionStatus;
