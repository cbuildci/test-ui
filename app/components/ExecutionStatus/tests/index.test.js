import React from 'react';
import { shallow, render } from 'enzyme';
import { IntlProvider } from 'react-intl';
import ExecutionStatus, { statusDisplay } from '../index';

const snapshots = (jsx) => {
    expect(shallow(jsx)).toMatchSnapshot();
    expect(render(
        <IntlProvider locale="en" initialNow={1500000100000}>{jsx}</IntlProvider>
    )).toMatchSnapshot();
};

describe('<ExecutionStatus />', () => {

    it('should have expected statuses for statusDisplay', () => {
        expect(Object.keys(statusDisplay).sort())
            .toEqual([
                '',
                'SUCCEEDED',
                'QUEUED',
                'IN_PROGRESS',
                'ERROR',
                'FAILED',
                'TIMED_OUT',
                'STOPPED',
                'NEUTRAL',
            ].sort());
    });

    for (const status of Object.keys(statusDisplay)) {
        it(`should render expected JSX for status ${JSON.stringify(status)}`, () => {
            snapshots(
                <ExecutionStatus status={status}/>,
            );
        });
    }

    it('should render expected JSX for unknown status', () => {
        snapshots(
            <ExecutionStatus status="foobar"/>,
        );

        snapshots(
            <ExecutionStatus/>,
        );
    });

});
