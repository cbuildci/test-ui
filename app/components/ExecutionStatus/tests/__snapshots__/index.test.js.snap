// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExecutionStatus /> should render expected JSX for status "" 1`] = `
<span
  style={
    Object {
      "color": "#333333",
    }
  }
>
  <i
    className="fas fa-question-circle fa-fw mr-1"
  />
  -
</span>
`;

exports[`<ExecutionStatus /> should render expected JSX for status "" 2`] = `
<span
  style="color:#333333"
>
  <i
    class="fas fa-question-circle fa-fw mr-1"
  />
  -
</span>
`;

exports[`<ExecutionStatus /> should render expected JSX for status "ERROR" 1`] = `
<span
  style={
    Object {
      "color": "#dc3545",
    }
  }
>
  <i
    className="fas fa-exclamation-triangle fa-fw mr-1"
  />
  <FormattedMessage
    defaultMessage="{status, select,
            QUEUED {Queued}
            IN_PROGRESS {In Progress}
            SUCCEEDED {Succeeded}
            ERROR {Internal System Error}
            FAILED {Failed}
            STOPPED {Stopped by User}
            TIMED_OUT {Timed Out}
            NEUTRAL {Neutral}
            other {{status}}
        }"
    id="app.components.ExecutionStatus.executionStatus"
    values={
      Object {
        "status": "ERROR",
      }
    }
  />
</span>
`;

exports[`<ExecutionStatus /> should render expected JSX for status "ERROR" 2`] = `
<span
  style="color:#dc3545"
>
  <i
    class="fas fa-exclamation-triangle fa-fw mr-1"
  />
  <span>
    Internal System Error
  </span>
</span>
`;

exports[`<ExecutionStatus /> should render expected JSX for status "FAILED" 1`] = `
<span
  style={
    Object {
      "color": "#dc3545",
    }
  }
>
  <i
    className="fas fa-times fa-fw mr-1"
  />
  <FormattedMessage
    defaultMessage="{status, select,
            QUEUED {Queued}
            IN_PROGRESS {In Progress}
            SUCCEEDED {Succeeded}
            ERROR {Internal System Error}
            FAILED {Failed}
            STOPPED {Stopped by User}
            TIMED_OUT {Timed Out}
            NEUTRAL {Neutral}
            other {{status}}
        }"
    id="app.components.ExecutionStatus.executionStatus"
    values={
      Object {
        "status": "FAILED",
      }
    }
  />
</span>
`;

exports[`<ExecutionStatus /> should render expected JSX for status "FAILED" 2`] = `
<span
  style="color:#dc3545"
>
  <i
    class="fas fa-times fa-fw mr-1"
  />
  <span>
    Failed
  </span>
</span>
`;

exports[`<ExecutionStatus /> should render expected JSX for status "IN_PROGRESS" 1`] = `
<span
  style={
    Object {
      "color": "#17a2b8",
    }
  }
>
  <i
    className="fas fa-hourglass-start fa-fw mr-1"
  />
  <FormattedMessage
    defaultMessage="{status, select,
            QUEUED {Queued}
            IN_PROGRESS {In Progress}
            SUCCEEDED {Succeeded}
            ERROR {Internal System Error}
            FAILED {Failed}
            STOPPED {Stopped by User}
            TIMED_OUT {Timed Out}
            NEUTRAL {Neutral}
            other {{status}}
        }"
    id="app.components.ExecutionStatus.executionStatus"
    values={
      Object {
        "status": "IN_PROGRESS",
      }
    }
  />
</span>
`;

exports[`<ExecutionStatus /> should render expected JSX for status "IN_PROGRESS" 2`] = `
<span
  style="color:#17a2b8"
>
  <i
    class="fas fa-hourglass-start fa-fw mr-1"
  />
  <span>
    In Progress
  </span>
</span>
`;

exports[`<ExecutionStatus /> should render expected JSX for status "NEUTRAL" 1`] = `
<span
  style={
    Object {
      "color": "#6c757d",
    }
  }
>
  <i
    className="fas fa-share fa-fw mr-1"
  />
  <FormattedMessage
    defaultMessage="{status, select,
            QUEUED {Queued}
            IN_PROGRESS {In Progress}
            SUCCEEDED {Succeeded}
            ERROR {Internal System Error}
            FAILED {Failed}
            STOPPED {Stopped by User}
            TIMED_OUT {Timed Out}
            NEUTRAL {Neutral}
            other {{status}}
        }"
    id="app.components.ExecutionStatus.executionStatus"
    values={
      Object {
        "status": "NEUTRAL",
      }
    }
  />
</span>
`;

exports[`<ExecutionStatus /> should render expected JSX for status "NEUTRAL" 2`] = `
<span
  style="color:#6c757d"
>
  <i
    class="fas fa-share fa-fw mr-1"
  />
  <span>
    Neutral
  </span>
</span>
`;

exports[`<ExecutionStatus /> should render expected JSX for status "QUEUED" 1`] = `
<span
  style={
    Object {
      "color": "#6c757d",
    }
  }
>
  <i
    className="far fa-clock fa-fw mr-1"
  />
  <FormattedMessage
    defaultMessage="{status, select,
            QUEUED {Queued}
            IN_PROGRESS {In Progress}
            SUCCEEDED {Succeeded}
            ERROR {Internal System Error}
            FAILED {Failed}
            STOPPED {Stopped by User}
            TIMED_OUT {Timed Out}
            NEUTRAL {Neutral}
            other {{status}}
        }"
    id="app.components.ExecutionStatus.executionStatus"
    values={
      Object {
        "status": "QUEUED",
      }
    }
  />
</span>
`;

exports[`<ExecutionStatus /> should render expected JSX for status "QUEUED" 2`] = `
<span
  style="color:#6c757d"
>
  <i
    class="far fa-clock fa-fw mr-1"
  />
  <span>
    Queued
  </span>
</span>
`;

exports[`<ExecutionStatus /> should render expected JSX for status "STOPPED" 1`] = `
<span
  style={
    Object {
      "color": "#dc3545",
    }
  }
>
  <i
    className="fas fa-stop-circle fa-fw mr-1"
  />
  <FormattedMessage
    defaultMessage="{status, select,
            QUEUED {Queued}
            IN_PROGRESS {In Progress}
            SUCCEEDED {Succeeded}
            ERROR {Internal System Error}
            FAILED {Failed}
            STOPPED {Stopped by User}
            TIMED_OUT {Timed Out}
            NEUTRAL {Neutral}
            other {{status}}
        }"
    id="app.components.ExecutionStatus.executionStatus"
    values={
      Object {
        "status": "STOPPED",
      }
    }
  />
</span>
`;

exports[`<ExecutionStatus /> should render expected JSX for status "STOPPED" 2`] = `
<span
  style="color:#dc3545"
>
  <i
    class="fas fa-stop-circle fa-fw mr-1"
  />
  <span>
    Stopped by User
  </span>
</span>
`;

exports[`<ExecutionStatus /> should render expected JSX for status "SUCCEEDED" 1`] = `
<span
  style={
    Object {
      "color": "#28a745",
    }
  }
>
  <i
    className="fas fa-check fa-fw mr-1"
  />
  <FormattedMessage
    defaultMessage="{status, select,
            QUEUED {Queued}
            IN_PROGRESS {In Progress}
            SUCCEEDED {Succeeded}
            ERROR {Internal System Error}
            FAILED {Failed}
            STOPPED {Stopped by User}
            TIMED_OUT {Timed Out}
            NEUTRAL {Neutral}
            other {{status}}
        }"
    id="app.components.ExecutionStatus.executionStatus"
    values={
      Object {
        "status": "SUCCEEDED",
      }
    }
  />
</span>
`;

exports[`<ExecutionStatus /> should render expected JSX for status "SUCCEEDED" 2`] = `
<span
  style="color:#28a745"
>
  <i
    class="fas fa-check fa-fw mr-1"
  />
  <span>
    Succeeded
  </span>
</span>
`;

exports[`<ExecutionStatus /> should render expected JSX for status "TIMED_OUT" 1`] = `
<span
  style={
    Object {
      "color": "#dc3545",
    }
  }
>
  <i
    className="fas fa-stopwatch fa-fw mr-1"
  />
  <FormattedMessage
    defaultMessage="{status, select,
            QUEUED {Queued}
            IN_PROGRESS {In Progress}
            SUCCEEDED {Succeeded}
            ERROR {Internal System Error}
            FAILED {Failed}
            STOPPED {Stopped by User}
            TIMED_OUT {Timed Out}
            NEUTRAL {Neutral}
            other {{status}}
        }"
    id="app.components.ExecutionStatus.executionStatus"
    values={
      Object {
        "status": "TIMED_OUT",
      }
    }
  />
</span>
`;

exports[`<ExecutionStatus /> should render expected JSX for status "TIMED_OUT" 2`] = `
<span
  style="color:#dc3545"
>
  <i
    class="fas fa-stopwatch fa-fw mr-1"
  />
  <span>
    Timed Out
  </span>
</span>
`;

exports[`<ExecutionStatus /> should render expected JSX for unknown status 1`] = `
<span
  style={
    Object {
      "color": "#333333",
    }
  }
>
  <i
    className="fas fa-question-circle fa-fw mr-1"
  />
  <FormattedMessage
    defaultMessage="{status, select,
            QUEUED {Queued}
            IN_PROGRESS {In Progress}
            SUCCEEDED {Succeeded}
            ERROR {Internal System Error}
            FAILED {Failed}
            STOPPED {Stopped by User}
            TIMED_OUT {Timed Out}
            NEUTRAL {Neutral}
            other {{status}}
        }"
    id="app.components.ExecutionStatus.executionStatus"
    values={
      Object {
        "status": "foobar",
      }
    }
  />
</span>
`;

exports[`<ExecutionStatus /> should render expected JSX for unknown status 2`] = `
<span
  style="color:#333333"
>
  <i
    class="fas fa-question-circle fa-fw mr-1"
  />
  <span>
    foobar
  </span>
</span>
`;

exports[`<ExecutionStatus /> should render expected JSX for unknown status 3`] = `
<span
  style={
    Object {
      "color": "#333333",
    }
  }
>
  <i
    className="fas fa-question-circle fa-fw mr-1"
  />
  -
</span>
`;

exports[`<ExecutionStatus /> should render expected JSX for unknown status 4`] = `
<span
  style="color:#333333"
>
  <i
    class="fas fa-question-circle fa-fw mr-1"
  />
  -
</span>
`;
